roofit/roofitcore/inc/RooFuncWrapper.h

```diff
@@ -124,14 +145,15 @@ class RooFuncWrapper final : public RooAbsReal {
    {
       updateGradientVarBuffer();
 
-      return _func(_gradientVarBuffer.data());
+      return _func(_gradientVarBuffer.data(), _observables.data());
    }
 
 private:
    RooListProxy _params;
    Func _func;
    Grad _grad;
    mutable std::vector<double> _gradientVarBuffer;
+   mutable std::vector<double> _observables;
```

Comment:
```suggestion
   std::vector<double> _observables;
```
This doesn't have to be mutable, right? It is set in the constructor.

Reply:
The `.data()` cannot be used in const member function for non-mutable members.

Reply:
I don't think that's correct, there is also a const version of `data()`:
https://en.cppreference.com/w/cpp/container/vector/data

If you change the signature of your function to take `double const*` instead of `double *` it should work.

Reply:
Anyway, that can be done in the next PR :+1: This one already has many changes